providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging/EasyAbp/NotificationService/Provider/PrivateMessaging/PrivateMessageNotificationSendingJob.cs

```diff
@@ -0,0 +1,69 @@
+﻿using EasyAbp.NotificationService.NotificationInfos;
+using EasyAbp.NotificationService.Notifications;
+using EasyAbp.PrivateMessaging.PrivateMessages;
+using JetBrains.Annotations;
+using System;
+using System.Threading.Tasks;
+using Volo.Abp.BackgroundJobs;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Timing;
+using Volo.Abp.Uow;
+
+namespace EasyAbp.NotificationService.Provider.PrivateMessaging
+{
+    public class PrivateMessageNotificationSendingJob : IAsyncBackgroundJob<PrivateMessageNotificationSendingJobArgs>, ITransientDependency
+    {
+        private readonly INotificationInfoRepository _notificationInfoRepository;
+        private readonly IUserUserNameProvider _userUserNameProvider;
+        private readonly INotificationRepository _notificationRepository;
+        private readonly IPrivateMessageAppService _privateMessageAppService;
+        private readonly IClock _clock;
+
+        public PrivateMessageNotificationSendingJob(
+            INotificationInfoRepository notificationInfoRepository,
+            IUserUserNameProvider userUserNameProvider,
+            INotificationRepository notificationRepository,
+            IPrivateMessageAppService privateMessageAppService,
+        IClock clock)
+        {
+            _notificationInfoRepository = notificationInfoRepository;
+            _userUserNameProvider = userUserNameProvider;
+            _notificationRepository = notificationRepository;
+            _privateMessageAppService = privateMessageAppService;
+            _clock = clock;
+        }
+
+        [UnitOfWork]
+        public virtual async Task ExecuteAsync(PrivateMessageNotificationSendingJobArgs args)
+        {
+            var notification = await _notificationRepository.GetAsync(args.NotificationId);
+
+            var userName = await _userUserNameProvider.GetAsync(notification.UserId);
+
+            if (userName.IsNullOrWhiteSpace())
+            {
+                await SaveNotificationResultAsync(notification, false, NotificationConsts.FailureReasons.ReceiverInfoNotFound);
+                return;
+            }
+
+            var notificationInfo = await _notificationInfoRepository.GetAsync(notification.NotificationInfoId);
+
+            await _privateMessageAppService.CreateAsync(new EasyAbp.PrivateMessaging.PrivateMessages.Dtos.CreateUpdatePrivateMessageDto
+            {
+                Title = notificationInfo.GetDataValue(NotificationProviderPrivateMessagingConsts.NotificationInfoTitlePropertyName).ToString(),
+                Content = notificationInfo.GetDataValue(NotificationProviderPrivateMessagingConsts.NotificationInfoContentPropertyName).ToString(),
+                ToUserName = userName
+            });
+
+            await SaveNotificationResultAsync(notification, true);
```

Comment:
Handle the [PrivateMessageSentEto](https://github.com/EasyAbp/PrivateMessaging/blob/e84cd76bf4f08cba0d804e03ce37daa184702d10/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/PrivateMessageSentEto.cs) event and set the notification result.

Reply:
We can not determine if the result of PM here by subscribing to PrivateMessageSentEto here. We will need something like a reference id when sending pm. By the info in PrivateMessageSentEto, we can only try to find notification by userid, but what if one user has send multiple messages at a time. @gdlcf88

Reply:
@slarkerino
OK, I will handle it :)

Reply:
@slarkerino

SendPrivateMessageEto is extensible now, see:
https://github.com/EasyAbp/PrivateMessaging/blob/master/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/SendPrivateMessageEto.cs

The `SendPrivateMessageEto.ExtraProperties` will be mapped to the `PrivateMessage.ExtraProperties`.

So after PM module v1.1.1, you can define a `NotificationId` property to the `PrivateMessage` entity to record the notification ID and get it from `PrivateMessageSentEto.ExtraProperties`.